src/Types/EnumType.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TheCodingMachine\GraphQLite\Types;
+
+use BackedEnum;
+use GraphQL\Type\Definition\EnumType as BaseEnumType;
+use UnitEnum;
+
+use function assert;
+use function is_string;
+
+/**
+ * An extension of the EnumType to support native enums.
+ */
+class EnumType extends BaseEnumType
+{
+    /** @var bool */
+    private $useValues;
+
+    /**
+     * @param class-string<UnitEnum> $enumName
+     */
+    public function __construct(string $enumName, string $typeName, bool $useValues = false)
+    {
+        $this->useValues = $useValues;
+
+        $values = [];
+        foreach ($enumName::cases() as $case) {
+            /** @var UnitEnum $case */
+            $values[$this->serialize($case)] = ['value' => $case];
+        }
+
+        parent::__construct([
+            'name' => $typeName,
+            'values' => $values,
+        ]);
+    }
+
+    // phpcs:disable SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+    // phpcs:disable SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint
+
+    /**
+     * @param UnitEnum $value
+     *
+     * @return string
+     */
+    public function serialize($value)
+    {
+        if (! $this->useValues) {
+            return $value->name;
+        }
+
+        assert($value instanceof BackedEnum);
+        assert(is_string($value->value));
```

Comment:
While assert is pretty cool, being a configurable option, I think it's best to throw Exceptions here.

Reply:
`EnumTypeMapper` actually validated these two assertions:

```php
        // Expose values instead of names if specifically configured to and if enum is string-backed
        $useValues = $typeAnnotation !== null &&
            $typeAnnotation->useValues() &&
            $reflectionEnum->isBacked() &&
            (string) $reflectionEnum->getBackingType() === 'string';
```

Unless we don't trust the core to input values according to the mapper? (in which case I should sign `mixed` and also make sure `$value instanceof UnitEnum` in the first place)

Reply:
That's a good point.  I wasn't aware checks were being done there.  I'd still argue that the requirements of one class shouldn't be reliant on another.  ideally we'd rely on proper type checking for this, but we're still supporting 7.2 at the moment.  We really need to bump the min support version to 7.4 IMO.

We can leave this - proper type checking should resolve in the future.